src/app/pages/datasets/components/dataset-form/sections/other-options-section/other-options-section.component.ts

```diff
@@ -355,26 +358,50 @@ export class OtherOptionsSectionComponent implements OnInit, OnChanges {
     }
   }
 
-  private setUpDedupWarnings(): void {
+  private setUpDedupWarning(): void {
     this.form.controls.deduplication.valueChanges.pipe(untilDestroyed(this)).subscribe((dedup) => {
       if (!dedup || [DeduplicationSetting.Off, inherit].includes(dedup)) {
-        this.hasDedupWarning = false;
         this.cdr.markForCheck();
         return;
       }
 
-      this.hasDedupWarning = true;
-      this.cdr.markForCheck();
+      this.dialogService.confirm({
+        title: this.translate.instant('Warning'),
+        message: this.translate.instant(helptextDatasetForm.deduplicationWarning),
+        hideCheckbox: true,
+      })
+        .pipe(untilDestroyed(this))
+        .subscribe((confirmed) => {
+          if (confirmed) {
+            this.checkDedupChecksum();
+          } else {
+            this.form.patchValue({
+              deduplication: inherit,
+            });
+          }
+        });
+    });
+  }
 
-      const checksum = this.form.controls.checksum.value;
-      if (this.wasDedupChecksumWarningShown || !checksum || checksum === DatasetChecksum.Sha512) {
-        return;
-      }
+  private checkDedupChecksum(): void {
+    const dedup = this.form.controls.deduplication.value;
+    if (!dedup || [DeduplicationSetting.Off, inherit].includes(dedup)) {
+      return;
+    }
 
-      this.showDedupChecksumWarning();
-      this.form.patchValue({
-        checksum: DatasetChecksum.Sha512,
-      });
+    const checksum = this.form.controls.checksum.value;
+    if (
+      this.wasDedupChecksumWarningShown
+      || !checksum
+      || checksum === DatasetChecksum.Sha512
+      || checksum !== DatasetChecksum.Sha256
```

Comment:
I expected the checksum warning to appear when dedupe is enabled and the checksum `Sha256` value is selected, but I haven't seen it yet. Is there something that I missed?

Reply:
Yeah, the logic is a bit incorrect here.